clang/lib/CodeGen/CGHLSLRuntime.cpp

```diff
@@ -423,6 +437,14 @@ void CGHLSLRuntime::emitEntryFunction(const FunctionDecl *FD,
   // FIXME: Handle codegen for return type semantics.
   // See: https://github.com/llvm/llvm-project/issues/57875
   B.CreateRetVoid();
+
+  // Add and identify root signature to function, if applicable
+  const AttrVec &Attrs = FD->getAttrs();
+  for (const Attr *Attr : Attrs) {
```

Comment:
I don't think naming `Attrs` is doing much here, probably simpler just to call the function in the loop header
```suggestion
  for (const Attr *Attr : FD->getAttrs()) {
```